InputManager.gd

```diff
@@ -69,37 +70,42 @@ func _unhandled_input(event):
 				var rel1 = event.position - last_mouse_press.position
 				var rel2 = rel1 + event.relative
 				emit("twist", InputEventScreenTwist.new({"position": last_mouse_press.position,
-													     "relative": rel1.angle_to(rel2),
+														 "relative": rel1.angle_to(rel2),
 														 "speed": event.speed}))
 	
 	# Touch.
 	elif event is InputEventScreenTouch:
 		if event.pressed:
 			touches[event.get_index()] = event 
 			if (event.get_index() == 0): # First and only touch.
-				emit("single_touch", InputEventSingleScreenTouch.new(event))
+				single_touch_cancelled = false
+				emit("single_touch", InputEventSingleScreenTouch.new(event, false))
 				only_touch = event
-				if tap_delay_timer.is_stopped(): tap_delay_timer.start(TAP_TIME_THRESHOLD)
+				if tap_delay_timer.is_stopped(): 
+					tap_delay_timer.start(TAP_TIME_THRESHOLD)
 			else:
-				only_touch = null
+				single_touch_cancelled = true
 				cancel_single_drag()
+				emit("single_touch", InputEventSingleScreenTouch.new(only_touch, true))
 		else:
 			touches.erase(event.get_index())
 			drags.erase(event.get_index())
 			cancel_single_drag()
-			if only_touch:
-				emit("single_touch", InputEventSingleScreenTouch.new(event))
+			if (only_touch and (event.get_index() == 0)):
+				emit("single_touch", InputEventSingleScreenTouch.new(event, single_touch_cancelled))
 				if !tap_delay_timer.is_stopped(): 
 					tap_delay_timer.stop()
 					emit("single_tap", InputEventSingleScreenTap.new(only_touch))
+				only_touch = null
 		
 	elif event is InputEventScreenDrag:
 		drags[event.index] = event
 		if !complex_gesture_in_progress():
 			if(drag_enabled):
 				emit("single_drag", InputEventSingleScreenDrag.new(event))
 			else:
-				if drag_startup_timer.is_stopped(): drag_startup_timer.start(DRAG_STARTUP_TIME)
```

Comment:
Please restore the one line if

Reply:
It is a common practise for any langage, for lisibility, to avoid such one lines writting.
It catches my eyes too much to let it like this ...